src/core.ts

```diff
@@ -51,18 +51,19 @@ const isTestOnlySymbol = (value: any): value is symbol => value === testOnlySymb
 /**
  * Perform the central validation/sanitization logic on the full environment object
  */
-export function getSanitizedEnv<T>(
+export function getSanitizedEnv<S>(
   environment: unknown,
-  specs: { [K in keyof T]: ValidatorSpec<T[K]> },
-  options: CleanOptions<T> = {},
-): T {
-  let cleanedEnv = {} as T
-  const errors: Partial<Record<keyof T, Error>> = {}
-  const varKeys = Object.keys(specs) as Array<keyof T>
+  specs: S,
+  options: CleanOptions<FromSpecsRecord<S>> = {},
+): FromSpecsRecord<S> {
+  let cleanedEnv = {} as Record<keyof S, unknown>
+  const castSpecs = specs as unknown as Record<keyof S, ValidatorSpec<unknown>>
```

Comment:
Naming nit– `castedSpecs` might be a little bit clearer

Reply:
9741e97a676d504599ed940b1e79a6ddc09ba82f